js/popup.js

```diff
@@ -1,423 +1,396 @@
+    var datetimeString = $("#add-addmeasurement-variable-date").val();
+
+    //alert (Date.parse(datetimeString)) ;
+    datetimeString = datetimeString.replace("AM", "");
+    datetimeString = datetimeString.replace("PM", "");
+    //
+
+
+    //year, month, day, hours, minutes, seconds, milliseconds
+    var hour = $('#add-addmeasurement-variable-timeh').val();
+    var min = $('#add-addmeasurement-variable-timem').val();
+    var ap = $('#add-addmeasurement-variable-timeap').val();
+    var datetime = new Date(datetimeString);
+
+    /*
+     datetime.setHours(parseInt(hour) + (ap * 12));
+     datetime.setMinutes(min);
+     datetime.setSeconds(0);
+     */
+
+    if (name == '') {
+        alert("Please enter the variable name.");
+        return;
+    }
+    if (value == '') {
+        alert("Please enter the value.");
+        return;
+    }
+    //var variable = getVariableWithName(name);
+    var measurements = [
+        {
+            timestamp: Math.floor(datetime.getTime() / 1000),
+            value: value
+        }
+    ];
+    // Add it to a request, payload is what we'll send to QuantiModo
+    var request = {
+        message: "uploadMeasurements",
+        payload: [
+            {
+                measurements: measurements,
+                name: name,
+                source: "QuantiModo",
+                category: valueCategory,
+                combinationOperation: combineOp,
+                unit: unit
+            }
+        ]
+
+    };
+    chrome.extension.sendMessage(request, function (responseText) {
+        var response = $.parseJSON(responseText);
+        if (response.success == true) {
+            document.getElementById('addmeasurement-variable-value').value = "";
+            document.getElementById('add-addmeasurement-variable-value').value = "";
+            window.close();
+        }
+        else {
+            alert("Adding a measurement failed.");
+            console.log(responseText);
+        }
+    });
+
+    clearNotifications();
 };
 
-var onVariableNameInputFocussed = function()
-{
-		$("#snd_gap").height('100px');
-	//document.getElementById('sectionMeasurementInput').style.opacity="0.2";
+var onVariableNameInputFocussed = function () {
+    $("#snd_gap").height('100px');
+    //document.getElementById('sectionMeasurementInput').style.opacity="0.2";
 };
-var onVariableNameInputUnfocussed = function()
-{
-	//$("#snd_gap").height('10px');
-	//document.getElementById('sectionMeasurementInput').style.opacity="1";
+var onVariableNameInputUnfocussed = function () {
+    //$("#snd_gap").height('10px');
+    //document.getElementById('sectionMeasurementInput').style.opacity="1";
 };
 
-var getVariableWithName = function(variableName)
-{
-	var filteredVars = jQuery.grep(variables, function (variable, i) {
-			return variable.name == variableName;
-	});
-	if (filteredVars.length > 0) return filteredVars[0];
-	return null;
+var getVariableWithName = function (variableName) {

```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['getVariableWithName' is defined but never used.](https://www.codacy.com/app/m_3/QuantiModo-Chrome-Extension/file/842132528/issues/source?bid=2779649&fileBranchId=2805440#l310)